frontend/apps/reader/modules/readersearch.lua

```diff
@@ -256,29 +255,60 @@ function ReaderSearch:onShowSearchDialog(text, direction, regex, case_insensitiv
         -- Keep the LTR order of |< and >|:
         from_start_text, from_end_text = BD.ltr(from_end_text), BD.ltr(from_start_text)
     end
+    self.wait_button = ButtonDialog:new{
+        buttons = {{{ text = "⌛" }}},
+    }
+    local function do_search_edge(func, text, regex, case_insensitive)
+        if regex then
+            return function()
+                self.wait_button.alpha = 0.75
+                UIManager:show(self.wait_button)
+                UIManager:tickAfterNext(function()
+                    do_search(func, text, regex, case_insensitive)()
+                    UIManager:close(self.wait_button)
+                end)
+            end
+        else
+            return do_search(func, text, regex, case_insensitive)
+        end
+    end
+    local function do_search_mid(func, text, param, regex, case_insensitive)
```

Comment:
`do_search_mid` and `do_search_edge` look quite similar. You made two because there is the `param` parameter in the middle in one and not the other? Then may be move it at the end, or pass a dummy value in those that don't use it.